src/auth0-provider-with-history.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Auth0Provider } from '@auth0/auth0-react';
+import { useHistory } from 'react-router-dom';
+
+const Auth0providerWithHistory = ({ children }) => {
+    const history = useHistory();
+    const domain = process.env.REACT_APP_AUTHO_DOMAIN;
+    const clientId = process.env.REACT_APP_AUTHO_CLIENT_ID;
+
+    const onRedirectCallback = (appState) => history.push(appState?.returnTo || window.location.pathname);
```

Comment:
shouold not be using window.location.pathname here....shoul use the `useRouter` hook and grab location from that

Reply:
In this case i actually would prefer to leave the or statement off, i would rather the app return to the main page when a user logs out of the app, which will happen no matter what because of my routing on the app page. I have just removed the or statement.

Reply:
if you want me to do this to learn hooks i can. i found nothing about a 'useRouter' hook, only things about the react router dom hooks, useHistory, useLocation, etc..